appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmRuntimeFileLogHandler.java

```diff
@@ -31,22 +31,20 @@
  * with the cloud logging agent.
  *
  */
-public class VmRuntimeFileLogHandler extends FileHandler {
-
+public class VmRuntimeFileLogHandler extends FileHandler implements SystemLogger {
   // This exists for testing purposes only.  If set, the cloud logger may lose logs.
   public static final String LOG_DIRECTORY_PROPERTY = "com.google.apphosting.logs";
   public static final String LOG_PATTERN_CONFIG_PROPERTY =
       "com.google.apphosting.vmruntime.VmRuntimeFileLogHandler.pattern";
   // Log files to /var/log/app_engine/app.[0-2].log.json
   private static final String DEFAULT_LOG_DIRECTORY = "/var/log/app_engine";
-  private static final String DEFAULT_LOG_PATTERN = "app.%g.log.json";
+  private static final String DEFAULT_LOG_PATTERN = "app.%u.%g.log.json";
   private static final String APP_ENGINE_LOG_CONFIG_PATTERN_ENV = "APP_ENGINE_LOG_CONFIG_PATTERN";
-  private static final int LOG_MAX_SIZE = 100 * 1024 * 1024;
+  private static final int LOG_PER_FILE_SIZE = 100 * 1024 * 1024;
```

Comment:
@joakime isn't there a way that we can get arbitary parameters out of the logging.properties file.   I think we briefly discussed this so that you could set something like `java.util.logging.FileHandler.fileSize=12345`